src/utils.jl

```diff
@@ -428,3 +428,33 @@ Get the amount dimension of a quantity (e.g., mol^(uamount)).
 """
 uamount(q::UnionAbstractQuantity) = uamount(dimension(q))
 uamount(d::AbstractDimensions) = d.amount
+
+"""
+Utility functions for `DynamicQuantitiesCollection`
+"""
+
+for f in (:enumerate, :length, :lastindex)
+    @eval begin
+        Base.$f(u::DynamicQuantitiesCollection) = $f(u.collection)
+    end
+end
+
+Base.getindex(u::DynamicQuantitiesCollection, i::Union{Int,  Symbol}) = getindex(u.collection, i)
+Base.getindex(u::DynamicQuantitiesCollection, i::UInt16) = getindex(u.collection, convert(Int, i))
+
+Base.setindex!(u::DynamicQuantities.DynamicQuantitiesCollection{Dict{Symbol, T}}, i::Int64, s::Symbol) where T <: Integer = setindex!(u.collection, i, s)
```

Comment:
Should throw an error if the key already exists

Reply:
I've added a check here https://github.com/SymbolicML/DynamicQuantities.jl/pull/107/commits/54ee0f8d27f1bd89dfa237329449067a421004c5#diff-47c27891e951c8cd946b850dc2df31082624afdf57446c21cb6992f5f4b74aa2R459